packages/cosmic-swingset/src/shutdown.js

```diff
@@ -57,9 +58,9 @@ export const makeFreshShutdown = () => {
 };
 
 let cachedShutdown = null;
-export const makeCachedShutdown = () => {
+export const makeCachedShutdown = (...args) => {
```

Comment:
The problem is that the args of the second call may not match the args of the first call that is cached, and would be silently ignored. I don't have a good answer here. I'd almost prefer we set a flag and don't allow making shutdown twice.

Reply:
> The problem is that the args of the second call may not match the args of the first call that is cached, and would be silently ignored.

SInce it's verbosity only, I'm not concerned.  I will add a comment.

Reply:
For Endo workers, terminate just rejects a promise, sets off the grace period promise, then cleanup follows. I think this might be a good model in general.

Reply:
I am not the one to write it (there's a reason `shutdown.js` is part of `cosmic-swingset`), but I'm anticipating (though not holding my breath for) the reusable library of child- and process-management tools that would have a better solution.